compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -2620,16 +2620,15 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
         let mut scope = &*self.scope;
         let hir_id = loop {
             match scope {
-                Scope::Binder { hir_id, allow_late_bound, .. } => {
-                    if *allow_late_bound {
-                        break *hir_id;
-                    } else {
-                        self.tcx.sess.delay_span_bug(
-                            rustc_span::DUMMY_SP,
-                            "Late bounds are not allowed for the given scope",
-                        );
-                        return;
-                    }
+                Scope::Binder { hir_id, allow_late_bound: true, .. } => {
+                    break *hir_id;
+                }
+                Scope::Binder { allow_late_bound: false, .. } => {
```

Comment:
This can just be in the same match arm as `Scope::ObjectLifetimeDefault` etc.

Reply:
How ? `Scope::ObjectLifetimeDefault` does a completely different thing, that would cause an infinite loop I think.

Reply:
Oops, I mean `Root`/`Body`